cmd/prepare-storage.go

```diff
@@ -69,21 +78,50 @@ func formatXLMigrateLocalEndpoints(endpoints EndpointList) error {
 
 // Cleans up tmp directory of local disks.
 func formatXLCleanupTmpLocalEndpoints(endpoints EndpointList) error {
-	for _, endpoint := range endpoints {
+	g := errgroup.WithNErrs(len(endpoints))
+	for index, endpoint := range endpoints {
 		if !endpoint.IsLocal {
 			continue
 		}
-		formatPath := pathJoin(endpoint.Path, minioMetaBucket, formatConfigFile)
-		if _, err := os.Stat(formatPath); err != nil {
-			if os.IsNotExist(err) {
-				continue
+		index := index
+		g.Go(func() error {
+			epPath := endpoints[index].Path
+			formatPath := pathJoin(epPath, minioMetaBucket, formatConfigFile)
+			if _, err := os.Stat(formatPath); err != nil {
```

Comment:
suggestion:
```go
	formatPath := pathJoin(epPath, minioMetaBucket, formatConfigFile)
	if _, err := os.Stat(formatPath); !os.IsNotExist(err) {
            return err
	} 
```

Reply:
I want to check that only `if err != nil` . The change that you suggested would not handle err == nil properly.

Reply:
> I want to check that only `if err != nil` . The change that you suggested would not handle err == nil properly.

Ah, yes.